src/utils/index.ts

```diff
@@ -25,52 +25,52 @@ export const applyValue = (obj: any, path: (string | number)[], value: any) => {
 
 // case 1: you only render with a single component
 export function createDataType<ValueType = unknown> (
-  is: (value: unknown) => boolean,
+  is: (value: ValueType) => boolean,
```

Comment:
I think we don't guarantee the value is `ValueType.` value must be one of `any` or `unknown`